test/common/http/alternate_protocols_test.cc

```diff
@@ -0,0 +1,87 @@
+#include "common/http/alternate_protocols.h"
+
+#include "test/test_common/simulated_time_system.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Http {
+
+namespace {
+class AlternateProtocolsTest : public testing::Test, public Event::TestUsingSimulatedTime {
+public:
+  AlternateProtocolsTest() : protocols_(simTime()) {}
+
+  AlternateProtocols protocols_;
+  const std::string hostname1_ = "hostname1";
```

Comment:
nit: it seems odd that these are members, I would have expected them as constants in an unnamed namespace in this file

Reply:
It's definitely odd! :) The google style guide prohibits non-POD global variables (including std::string) because of static initializer issues. It looks like the Envoy style guide is a bit more flexible:

It says this:

> The Google C++ style guide points out that non-PoD static and global variables are forbidden. This includes types such as std::string. We encourage the use of the advice in the C++ FAQ on the static initialization fiasco for how to best handle this.

That doc recommends: 

> the Construct On First Use Idiom: replace the namespace-scope / global Fred object x with a namespace-scope / global function x() that returns the Fred object by reference.

This is definitely an option here but it seems more odd than use members. WDYT?

Reply:
Ah, right. Functions are a good solution but what you have now is fine.